### ISSUE TITLE
Unexpecte behaviour when selecting Website Data

### ISSUE THREAD

**Issue by SimonBasca** [mozilla-mobile/firefox-ios#8863]
## Steps to reproduce
1. Go to Data Management -> Website Data
2. Type "g"
3. Try selecting the website data that starts with "g" letter
### Expected behavior
- The data should be selected.
### Actual behavior
- The website data is not selected. After going clearing the search field different website data is selected.
### Device & build information
* Device: iPhone Xr (14.6)
* Build version: 35.0 (5448)
### Notes
Attachments: 
- [video.mp4.zip](https://github.com/mozilla-mobile/firefox-ios/files/6829025/video.mp4.zip)

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/FXIOS-2968)

**Comment by gutley** [comment]
WebsiteDataSearchResultsViewController is incorrectly using viewModel.siteRecords instead of filteredSiteRecords when changing the selection.

This patch fixes it:

```
diff --git a/Client/Frontend/Settings/WebsiteDataSearchResultsViewController.swift b/Client/Frontend/Settings/WebsiteDataSearchResultsViewController.swift
index 70916d8ce..d5acd6b88 100644
--- a/Client/Frontend/Settings/WebsiteDataSearchResultsViewController.swift
+++ b/Client/Frontend/Settings/WebsiteDataSearchResultsViewController.swift
@@ -97,7 +97,7 @@ class WebsiteDataSearchResultsViewController: UIViewController, UITableViewDataS
         let section = Section(rawValue: indexPath.section)!
         switch section {
         case .sites:
-            guard let item = viewModel.siteRecords[safe: indexPath.row] else { return }
+            guard let item = filteredSiteRecords[safe: indexPath.row] else { return }
             viewModel.selectItem(item)
             break
         case .clearButton:
@@ -112,7 +112,7 @@ class WebsiteDataSearchResultsViewController: UIViewController, UITableViewDataS
         let section = Section(rawValue: indexPath.section)!
         switch section {
         case .sites:
-            guard let item = viewModel.siteRecords[safe: indexPath.row] else { return }
+            guard let item = filteredSiteRecords[safe: indexPath.row] else { return }
             viewModel.deselectItem(item)
             break
         default: break;

```

**Comment by data-sync-user** [comment]
➤ Simion Basca commented:

Verifying as fix on main 6b01440f6274573f010b501a7c52de6d19c72bd3.

Note that while testing one more issue was discovered 
https://github.com/mozilla-mobile/firefox-ios/issues/10145 ( https://github.com/mozilla-mobile/firefox-ios/issues/10145|smart-link )
